src/index.ts

```diff
@@ -0,0 +1,104 @@
+import * as core from "@actions/core";
+import { File as DiffFile } from "parse-diff";
+import minimatch from "minimatch";
+import { callModel, createReviewPrompt, parseAIResponse } from "./ai";
+import { ReviewComment } from "./types";
+import {
+  getFileContent,
+  getPRContext,
+  getRepoContext,
+  submitReviewComments,
+} from "./git";
+
+async function prepareReviewContext(
+  prContext: Awaited<ReturnType<typeof getPRContext>>
+) {
+  const { pr, files, parsedDiff, owner, repo } = prContext;
+
+  const repoContext = await getRepoContext(owner, repo);
+  const excludePatterns = core
+    .getInput("exclude")
+    .split(",")
+    .map((s) => s.trim());
+  const filesWithContent = await Promise.all(
+    parsedDiff.map(async (file: DiffFile) => {
+      const filePath = file.to;
+      if (
+        !filePath ||
+        file.to === "/dev/null" ||
+        excludePatterns.some((pattern) => minimatch(filePath, pattern))
```

Comment:
The check for '/dev/null' is specific and may not cover all edge cases. Consider adding more robust checks for invalid file paths.